### PR TITLE
Do not mark ancestors of device with source or stage2 as protected

### PR DESCRIPTION
**Pull request by rvykydal**

### CODE REVIEW

**Review by jstodola — APPROVED** [review]
It looks OK to me.

**Review by vojtechtrefny — APPROVED** [review]

File: pyanaconda/modules/storage/devicetree/model.py [hunk 1]
```diff
@@ -285,6 +285,7 @@ def _mark_protected_devices(self):
         identify protected devices.
         """
         protected = []
+        protected_w_ancestors = []
```
Comment (jkonecny12) [inline pyanaconda/modules/storage/devicetree/model.py:288]: I wonder what `w` means here. Is it `with` or `without`? I would like to not use this abbreviations.

File: pyanaconda/modules/storage/devicetree/model.py [hunk 2]
```diff
@@ -353,23 +358,31 @@ def protect_devices(self, protected_names):
         # Update the list.
         self.protected_devices = protected_names
 
-    def _mark_protected_device(self, device):
+    def _mark_protected_device(self, device, ancestors=False):
```
Comment (jkonecny12) [inline pyanaconda/modules/storage/devicetree/model.py:361]: Could we please name the boolean to make it's obvious it's not a list of ancestors. I would suggest `protect_ancestors` or `include_ancestors`?

File: pyanaconda/modules/storage/devicetree/model.py [hunk 3]
```diff
@@ -353,23 +358,31 @@ def protect_devices(self, protected_names):
...
+        device.protected = True
+        log.debug("Marking device %s as protected.", device.name)
+        if ancestors:
+            for d in device.ancestors:
+                log.debug("Marking ancestor %s of device %s as protected.",
+                          d.name, device.name)
+                d.protected = True
 
-    def _mark_unprotected_device(self, device):
+    def _mark_unprotected_device(self, device, ancestors=False):
```
Comment (jkonecny12) [inline pyanaconda/modules/storage/devicetree/model.py:374]: The same here.

**Review by jkonecny12 — CHANGES_REQUESTED** [review]
Looks good to me. Just a nitpicks below.

However, please fix the `Resolves` in the commit message to point to RHEL issue and not BZ :).

File: pyanaconda/modules/storage/devicetree/model.py (hunk 1 above)
Comment (rvykydal) [reply pyanaconda/modules/storage/devicetree/model.py]: Updated

File: pyanaconda/modules/storage/devicetree/model.py (hunk 2 above)
Comment (rvykydal) [reply pyanaconda/modules/storage/devicetree/model.py]: Updated

File: pyanaconda/modules/storage/devicetree/model.py (hunk 3 above)
Comment (rvykydal) [reply pyanaconda/modules/storage/devicetree/model.py]: Updated

**Review by jkonecny12 — APPROVED** [review]
Looks good to me.

**Pull request merged**